### PR TITLE
op-e2e: Adopt op-chain-ops

### PR DESCRIPTION
**Pull request by mslipper**
This PR updates op-e2e to use op-chain-ops. Adopting op-chain-ops in op-e2e improves the following:

1. Transactions now route through the correct proxy contracts.
2. There's no need to wait for a contract deployment since conracts exist in genesis.
3. We can export the test helpers in op-e2e to enable us to run similar end-to-end tests in other applications like the indexer

### CODE REVIEW

**Comment by trianglesphere** [comment]
@mslipper I think the secrets have already been ported over to the `e2eutils` folder: https://github.com/ethereum-optimism/optimism/blob/develop/op-e2e/e2eutils/secrets.go

**Comment by mslipper** [comment]
oh nice! closing this then will restack.

**Review by tynes — APPROVED** [review]

**Comment by mslipper** [comment]
I rolled #3650 in here since the secrets util was already included - will push once tests pass locally.

**Comment by mslipper** [comment]
Removing dead code made this a hair over PR size SLA.

File: op-e2e/setup.go [hunk 1]
```diff
@@ -178,135 +236,78 @@ func (cfg SystemConfig) start() (*System, error) {
...
-	batchSubmitterAddr := crypto.PubkeyToAddress(bssPrivKey.PublicKey)
 
-	p2pSignerPrivKey, err := wallet.PrivateKey(accounts.Account{
-		URL: accounts.URL{
-			Path: cfg.P2PSignerHDPath,
-		},
-	})
-	if err != nil {
-		return nil, err
+	for addr, amount := range cfg.Premine {
```
Comment (tynes) [inline op-e2e/setup.go:244]: Is this premine still necessary? Can we dedupe this code and add something to the chain ops package and not set anything here?

Comment (mslipper) [reply op-e2e/setup.go:244]: It is - we sometimes want to override balances in the dev accounts, so having a way to do that without specifying all of them is nice.

We can certainly upgrade the chain ops package, but I'd rather minimize the diff here and do that separately.

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
being able to adopt this code and have it used as the test infra is nice, gives more confidence in using it for the actual state surgery

File: op-e2e/e2eutils/secrets.go [hunk 2]
```diff
@@ -43,8 +45,6 @@ type MnemonicConfig struct {
 	Mallory string
 }
 
-// Secrets computes the private keys for all mnemonic paths,
```
Comment (trianglesphere) [inline op-e2e/e2eutils/secrets.go:46]: why are these doc comments getting removed?

File: op-e2e/setup.go [hunk 3]
```diff
@@ -178,135 +236,78 @@ func (cfg SystemConfig) start() (*System, error) {
...
+		existing, ok := l1Genesis.Alloc[addr]
+		if ok {
```
Comment (trianglesphere) [inline op-e2e/setup.go:246]:
```suggestion
		if existing, ok := l1Genesis.Alloc[addr]; ok {
```

**Review by trianglesphere — COMMENTED** [review]
2 small things, but I'm excited for this

File: op-e2e/e2eutils/secrets.go (hunk 2 above)
Comment (mslipper) [reply op-e2e/e2eutils/secrets.go:46]: I coped this from Hive - can add it back.

File: op-e2e/e2eutils/secrets.go [hunk 4]
```diff
@@ -94,9 +101,9 @@ func (m *MnemonicConfig) Secrets() (*Secrets, error) {
 	}, nil
 }
 
-// Secrets bundles secp256k1 private keys for all common rollup actors for testing purposes.
 type Secrets struct {
```
Comment (trianglesphere) [inline op-e2e/e2eutils/secrets.go:104]:
```suggestion
// Secrets bundles secp256k1 private keys for all common rollup actors for testing purposes.
type Secrets struct {
```

File: op-e2e/e2eutils/secrets.go [hunk 5]
```diff
@@ -131,9 +139,9 @@ func (s *Secrets) Addresses() *Addresses {
 	}
 }
 
-// Addresses bundles the addresses for all common rollup addresses for testing purposes.
 type Addresses struct {
```
Comment (trianglesphere) [inline op-e2e/e2eutils/secrets.go:142]:
```suggestion
// Addresses bundles the addresses for all common rollup addresses for testing purposes.
type Addresses struct {
```

**Comment by tynes** [comment]
I'd like to get this merged so I can use it to write better tests for https://github.com/ethereum-optimism/optimism/pull/3528
Realizing that the tests that I'd like to write belong more in `op-e2e` than in `op-chain-ops`

**Comment by mslipper** [comment]
I'm debugging a test timeout. Stand by.

**Review by trianglesphere — APPROVED** [review]

**Comment by mslipper** [comment]
Ok, this one was really interesting - the timeout was caused by a deadlock in the `SyncStatus` function. Here's the implementation:

```
func (s *state) SyncStatus(ctx context.Context) (*eth.SyncStatus, error) {
	respCh := make(chan eth.SyncStatus)
	select {
	case <-ctx.Done():
		return nil, ctx.Err()
	case s.syncStatusReq <- respCh:
		select {
		case <-ctx.Done():
			return nil, ctx.Err()
		case resp := <-respCh:
			return &resp, nil
		}
	}
}
```

In some cases, the e2e test would cancel the passed-in context. This caused the function to return. If the timing as correct, the context would be canceled _before_ the driver loop had a chance to send a response to `respCh`. This caused closing the rollup node to hang, since the listener for the `done` channel in `Close()` is inside the big driver loop.

The solution was to make `respCh` buffered, so that the driver can still send responses to the channel after the `SyncStatus` context is closed.

**Pull request merged**